app/pages/learn/01_tutorial/01_your-first-java-app/02_building-with-intellij-idea.md

```diff
@@ -0,0 +1,207 @@
+
+[![Run](/assets/images/intellij-idea/run.png)](/assets/images/intellij-idea/run.png)
+
+If we want to pass arguments to our application, we can do so in our [Run Configurations](https://www.jetbrains.com/help/idea/run-debug-configuration.html).
+
+To edit your run configurations, select the configuration in the run/debug configuration switcher, by clicking the down arrow next to the current configuration or the three dots to the right of the run configuration, and select **Edit Configurations**.
+
+[![Edit Configurations](/assets/images/intellij-idea/edit-configurations.png)](/assets/images/intellij-idea/edit-configurations.png)
+
+Edit the **Run/Debug Configurations** in the popup.
+
+[![Run / Debug Configuration](/assets/images/intellij-idea/run-config.png)](/assets/images/intellij-idea/run-config.png)
+
+## Testing
+IntelliJ IDEA makes it easy to add tests to your code. You can navigate to the test for a particular class using the shortcut **⇧⌘T** on macOS or **Ctrl+Shift+T** on Windows/Linux. If no test class exists yet, IntelliJ IDEA will create one for you. This class will be created in the right location.
+We can select a **Testing library** in the **Create test** popup.
+
+[![Create test](/assets/images/intellij-idea/create-test.png)](/assets/images/intellij-idea/create-test.png)
+
+IntelliJ IDEA supports multiple testing libraries, including [JUnit5](https://junit.org/junit5/), which is the [most used testing library for Java developers](https://www.jetbrains.com/lp/devecosystem-2023/java/#java_unittesting).
+
+If JUnit5 is not part of your project yet, IntelliJ IDEA will note “JUnit5 library not found in the module”. Click **Fix** to have IntelliJ IDEA fix this for you.
+
+[![Create test - Fix](/assets/images/intellij-idea/create-test-fix.png)](/assets/images/intellij-idea/create-test-fix.png)
+
+Note that the JUnit5 dependency is added to the pom.xml.
+
+[![JUnit5 dependencies](/assets/images/intellij-idea/junit5-dependencies.png)](/assets/images/intellij-idea/junit5-dependencies.png)
+
+Go back to the test file to add tests. We can let IntelliJ IDEA help us generate our test for us. In the test class, we can use **Generate** (**⌘N** on macOS or **Alt+Insert** on Windows/Linux) and select **Test Method** to add a test. Give the test a name that explains the intended behavior, and add the relevant test code.
+
+In our test class, we can select **Run All Tests** (**⌃⇧R** on macOS or **Ctrl+Shift+F10** on Windows/Linux).
+While it’s nice to see our tests pass, we also want to make sure that they fail when something is wrong. Make some changes to your code that will make your tests fail, and run them again to see them fail. Revert those changes to see them pass again.
+
+We can view the code and tests side by side. To do so, click on the tab with the test file, and select **Move and Split right**. This can be especially helpful if you're doing [test-driven development (TDD)](https://martinfowler.com/bliki/TestDrivenDevelopment.html).
```

Comment:
I think beginners probably don't (need to) know about TDD. Many of them will probably not even touch unit tests for quite some time (when their course introduced libraries or whatever).

Reply:
Maybe one possibility would be to change the first paragraph of the introduction to something that makes it clear this isn't necessary for writing the first application.

After all, when I would tell someone to get started with an IDE, I would want them to be able to run the code, be informed about errors and maybe be able to use a debugger.
When I would tell them to follow that article, they might be confused about testing/Maven projects and not be able to set up anything/be overwhelmed with it.

Reply:
I'd be happy to add a brief explanation about build tools and testing (and why they're relevant for beginners to know about, hopefully without making it too overwhelming for them). But waiting for an answer from Ana-Maria on my other question first.

Reply:
@ammbra As discussed: Add brief explanation on why & how testing can help beginners (to break a problem into smaller pieces and test/verify that a small piece of code does what they expect it to do).